### PR TITLE
[platform_channels] adds ios implementation for MethodChannel demo

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description
Adds iOS implementation for MethodChannel Demo.

cc:// @RedBrogdon let me know if other file changes I pushed are meant to be pushed or not. I fairly new to iOS, so let me know if my swift implementation needs to be improved.

### CODE REVIEW

**Comment by AyushBherwani1998** [comment]
googlebot I fixed it.

File: platform_channels/ios/Runner/AppDelegate.swift [hunk 1]
```diff
@@ -3,11 +3,31 @@ import Flutter
...
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+    ) -> Bool {
+        let flutterViewController = window.rootViewController as! FlutterViewController
+        FlutterMethodChannel(name: "methodChannelDemo", binaryMessenger: flutterViewController.binaryMessenger).setMethodCallHandler({
+            (call: FlutterMethodCall, result: FlutterResult) -> Void in
+            
+            let count = (call.arguments as? NSDictionary)?["count"] as? Int
+            
+            if count == nil {
```
Comment (domesticmouse) [inline platform_channels/ios/Runner/AppDelegate.swift:16]: This probably should use `guard let`.

**Review by domesticmouse — APPROVED** [review]

File: platform_channels/ios/Runner/AppDelegate.swift (hunk 1 above)
Comment (AyushBherwani1998) [reply platform_channels/ios/Runner/AppDelegate.swift:16]: Something like this

```swift
guard let count = (call.arguments as? NSDictionary)?["count"] as? Int else {
    // Throw invalid argument error
}
```

Comment (AyushBherwani1998) [reply platform_channels/ios/Runner/AppDelegate.swift]: Done @domesticmouse

File: platform_channels/ios/Runner/AppDelegate.swift [hunk 2]
```diff
@@ -3,11 +3,30 @@ import Flutter
...
+        FlutterMethodChannel(name: "methodChannelDemo", binaryMessenger: flutterViewController.binaryMessenger).setMethodCallHandler({
+            (call: FlutterMethodCall, result: FlutterResult) -> Void in
+            
+            guard let count = (call.arguments as? NSDictionary)?["count"] as? Int else {
+                result(FlutterError(code: "INVALID_ARGUMENT", message: "Value of count cannot be null", details: nil))
+                return
+            }
+            
+            switch call.method {
+            case "increment":
```
Comment (RedBrogdon) [inline platform_channels/ios/Runner/AppDelegate.swift:20]: I did a double-take at the indentation here, but it's correct. 😄

**Review by RedBrogdon — APPROVED** [review]
LGTM. Thanks for putting that new laptop to work for Flutter. 😄

**Pull request merged**